### PR TITLE
Improve SSH key parser to handle newlines in keys

### LINKED ISSUE #7500: Adding a deploy key from copy/paste doesn't work
- Gitea version (or commit ref): 1.8.3

## Description

If I do a `cat ~/.ssh/id_rsa.pub` and copy paste it in Gitea to the textarea to add a new deploy key, I get the error "Can not verify your SSH key: extractTypeFromBase64Key: invalid key format: illegal base64 data at input byte 168"

If I remove the line break in the middle, it works.

It seems that Gitea doesn't handle line returns inside keys.

### PR DESCRIPTION
**Pull request by silverwind**
Reworked the logic to identify to detect PEM-encoded keys via a static string `-----BEGIN` and on regular SSH public key format, strip all newlines in the string which prevents extra newlines from being an issue on regular keys.

Fixes: https://github.com/go-gitea/gitea/issues/7500

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by techknowlogick** [comment]
Could you add some tests to `ssh_key_test.go`?

**Comment by silverwind** [comment]
Yes, will look into it. I already had a look earlier and was quite shocked that this function was completely untested.

**Comment by silverwind** [comment]
Tests added, ready for review.

I could not get PEM-encoded keys to work at all, the code that checks them was requesting absurd key lengths with an error `extractTypeFromBase64Key: invalid key format: not enough length 813826442` when I try to load a 2048 bit-encoded key, so these type of keys is probably completely broken (and untestet) currently.

File: models/ssh_key_test.go [hunk 1]
```diff
@@ -56,6 +56,27 @@ func Test_SSHParsePublicKey(t *testing.T) {
 	}
 }
 
+func Test_CheckPublicKeyString(t *testing.T) {
```
Comment (gary-kim) [inline models/ssh_key_test.go:59]: Just as a side comment, you can actually skip assigning to a variable as well if you'd like.
https://github.com/ncw/rclone/blob/8e8b78d7e57fae885a30df4d57eaa1da8f03dbd5/backend/sftp/sftp_internal_test.go#L39-L51

**Review by gary-kim — APPROVED** [review]
I think we can fix the issue with PEM encoded keys separately.

File: models/ssh_key_test.go (hunk 1 above)
Comment (silverwind) [reply models/ssh_key_test.go:59]: Thanks, simplified.

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/issues/7581 for the PEM issue.

**Comment by silverwind** [comment]
Accidentially closed this. Should be good to merge (and backport).

**Pull request merged**